### DISCUSSION TITLE
Relationship-based Access Control / FGA / Zanzibar

### DISCUSSION THREAD

**Discussion by StevenLangbroek** [supabase/supabase#12510] [Feature Requests]
Hey folks! We're just getting started with Supabase, and I was wondering if there's any appetite around fine-grained or relationship-based authorization, as first described by Google in their [Zanzibar paper](https://research.google/pubs/pub48190/). I've seen a few implementations in the wild:

- [Oso](https://www.osohq.com/)
- [OpenFGA](https://openfga.dev/)
- [Keto](https://github.com/ory/keto)

Would love to discuss :)

**Comment by adriangalilea** [comment]
Hey, I'm super interested in this.

I figured supabase is not the way to implement ReBAC perhaps, or at least not for now. But I'm intrigued ;)

As far as I've seen Keto has some implementation with supabase, but not sure how well they play together with what supabase already does in terms of auth/authz.

**Comment by mrkwtz** [comment]
I'd say doing something like with the Auth Hooks, where you instead hook into the authorization, define an endpoint for external authz check and return with 200 / 403.

**Comment by abeaclark** [comment]
Would LOVE an openFGA implementation. It supports postgres as a storage option already. A first-class integration into RLS would take supabase to the next level in terms of building enterprise grade applications.

**Comment by benschac** [comment]
I too would love this.
